src/experimental/queue.rs

```diff
@@ -0,0 +1,372 @@
+use async_std::sync::TryRecvError;
+use async_std::sync::{channel, Receiver, Sender};
+use log::warn;
+use std::collections::HashMap;
+use std::collections::VecDeque;
+use wasm_bindgen_futures::spawn_local;
+
+// BIG PICTURE
+//
+// Took aaron's idea: serialize processing and stack-allocate our
+// book-keeping so that ownership is clear and we can rely on
+// rust's compile-time borrow checking instead of runtime checking
+// and locks.
+// 
+// Dispatch happens in two layers. dispatch_loop is the first layer, 
+// responsible for opening and managing connections to databases. Each 
+// database ("store" in the code here) has its own connection_loop.
+// The dispatch_loop sends requests for a given db to its connection_loop.
+//
+// The connection_loop keeps a queue of incoming requests. It keeps at most
+// one write tx open at a time and when a write tx is open no other txs may run.
+// In order not to deadlock the connection_loop scans the queue for requests
```

Comment:
I considered this idea but could not wrap my head around whether it was correct to eagerly process messages for a write transaction ahead of other messages destined for other transactions.